pkg/clusterresource/controller.go

```diff
@@ -282,7 +282,8 @@ func (c *controller) Sync(ctx context.Context) error {
 	var errs = make([]error, 0)
 	for _, project := range projects {
 		for _, domain := range *domains {
-			namespace := fmt.Sprintf(namespaceFormat, project.Identifier, domain.Name)
+			namespaceMapping = c.config.NamespaceMappingConfiguration().GetNamespaceMappingConfig()
+			namespace := common.GetNamespaceName(namespaceMapping, project.Identifier, domain.Name)
```

Comment:
nit, no need to declare a new variable for the namespaceMapping, you can always just inline the GetNamespaceMappingConfig call

Reply:
Removed namespaceMapping variable.